transport/shadowsocks/stream.go

```diff
@@ -63,9 +63,9 @@ var (
 	_ io.ReaderFrom = (*Writer)(nil)
 )
 
-// NewShadowsocksWriter creates a [Writer] that encrypts the given [io.Writer] using
+// NewWriter creates a [Writer] that encrypts the given [io.Writer] using
 // the shadowsocks protocol with the given encryption key.
-func NewShadowsocksWriter(writer io.Writer, key *EncryptionKey) *Writer {
+func NewWriter(writer io.Writer, key *EncryptionKey) *Writer {
```

Comment:
The `Writer` type is inconsistent with the following `Reader` type, one is a `struct` while the other is an `interface`. Since both are exported types, I think this might confuse the caller.

Reply:
Good point! I'll add that to my list of things to refine.